cfme/dashboard.py

```diff
@@ -1,130 +1,365 @@
+            @classmethod
+            def all(cls, browser):
+                return [(browser.text(e), ) for e in browser.elements(cls.ALL_LOCATOR)]
+
     @property
     def is_displayed(self):
         return (
             self.logged_in_as_current_user and
             self.navigation.currently_selected == ['Cloud Intel', 'Dashboard'])
 
 
-def click_top_right(item):
-    base_locator = '//nav//a[@id="dropdownMenu2"]/../ul//a[normalize-space(.)="{}"]'
-    sel.click(page.user_dropdown)
-    sel.click(base_locator.format(item), wait_ajax=False)
-    sel.handle_alert(wait=False)
+class ParticularDashboardView(DashboardView):
+    @property
+    def is_displayed(self):
+        return (
+            super(ParticularDashboardView, self).is_displayed and
+            self.dashboards(title=self.obj.name).is_active)
+
+
+class DashboardCollection(Navigatable):
+    """Represents the Dashboard page and can jump around various dashboards present."""
+    @property
+    def default(self):
+        """Returns an instance of the ``Default Dashboard``"""
+        return self.instantiate('Default Dashboard')
+
+    def instantiate(self, name):
+        return Dashboard(name, collection=self)
+
+    def all(self):
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        result = []
+        # TODO: Idiomatize the following line
+        for (dashboard_name, ) in view.dashboards.view_class.all(view.browser):
+            result.append(self.instantiate(dashboard_name))
+        return result
+
+    def refresh(self):
+        """Refreshes the dashboard view by forcibly clicking the navigation again."""
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        view.navigation.select('Cloud Intel', 'Dashboard')
+
+    @property
+    def zoomed_name(self):
+        """Grabs the name of the currently zoomed widget."""
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        if not view.zoomed.is_displayed:
+            return None
+        return view.zoomed.title.text
+
+    def close_zoom(self):
+        """Closes any zoomed widget."""
+        navigate_to(self.appliance.server, 'Dashboard').ensure_zoom_closed()
+
+
+class Dashboard(Navigatable):
+    def __init__(self, name, collection=None, appliance=None):
+        if collection is None:
+            collection = DashboardCollection(appliance=appliance)
+        self.collection = collection
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.name = name
+
+    @property
+    def dashboard_view(self):
+        """Returns a view pointed at a particular dashboard."""
+        return navigate_to(self, 'Details').dashboards(title=self.name)
+
+    def reset_widgets(self, cancel=False):
+        """Clicks the Reset widgets button."""
+        view = navigate_to(self, 'Details')
+        view.reset_widgets()
+
+    def instantiate(self, name):
+        return DashboardWidget(name, self)
+
+    def all(self, content_type=None):  # widgets
```

Comment:
Your comment says is all - `widgets`. I would prefer if this was renamed to `widgets` and either give you `DashboardWidgetCollection` through it's `all` property or directly give you the DashboardWidgets as it does now. If you do decide to create `DashboardWidgetCollection`, it might make sense to move `reset_widgets` into it as `reset` but I don't recall this part of the UI so well so you tell me if that makes sense.

Anyway, `all` makes sense when we have context provided (in the class name for example, as we do with collections); in this case, the context is `Dashboard` and it's not correct.